modules/gateway/peers.go

```diff
@@ -184,62 +177,79 @@ func (g *Gateway) managedAcceptConnNewPeer(conn net.Conn, remoteVersion string)
 		return err
 	}
 
+	// Attempt to add the peer to the node list. If the add is successful and
+	// the address is a local address, mark the peer as a local peer.
+	local := false
+	err = g.managedAddUntrustedNode(remoteAddr)
+	if err != nil && remoteAddr.IsLocal() {
+		local = true
```

Comment:
According to [this answer](http://security.stackexchange.com/questions/37481/is-it-possible-to-pass-tcp-handshake-with-spoofed-ip-address), you can only get away with spoofing your IP if you control a router "close to the target." Of course, if you control such a router, you can probably do much worse things than that. But there's no harm in guarding against it unless it severely impacts other code.